pdb2sql/pdb2sql_base.py

```diff
@@ -127,27 +130,95 @@ def exportpdb(self, fname, append=False, periodic=False, **kwargs):
-            line += '{:^4}'.format(d[1])    # name
+            line += self._format_atomname(d) # name
             line += '{:>1}'.format(d[2])    # altLoc
-            line += '{:>3}'.format(d[3])  # resname
+            line += '{:>3}'.format(d[3])    # resname
             line += ' '
             line += '{:>1}'.format(d[4])    # chainID
             line += '{:>4}'.format(d[5])    # resSeq
             line += '{:>1}'.format(d[6])    # iCODE
             line += '   '
-            line += '{: 8.3f}'.format(d[7])  # x
-            line += '{: 8.3f}'.format(d[8])  # y
-            line += '{: 8.3f}'.format(d[9])  # z
+            line += pdb2sql_base._format_xyz(d[7]) # x
+            line += pdb2sql_base._format_xyz(d[8]) # y
+            line += pdb2sql_base._format_xyz(d[9])  # z
             if not self.no_extra:
-                line += '{: 6.2f}'.format(d[10])    # occ
-                line += '{: 6.2f}'.format(d[11])    # temp
+                line += '{:>6.2f}'.format(d[10])    # occ
+                line += '{:>6.2f}'.format(d[11])    # temp
+                line += ' ' * 10
+                line += '{:>2}'.format(d[12])       # element
             line += '\n'
 
             f.write(line)
 
         # close
         f.close()
 
+    def _format_atomname(self, data):
+        """Format atom name to align with PDB reqireuments:
+             - alignment of one-letter atom name starts at column 14,
+             - while two-letter atom name such as FE starts at column 13.
+
+        Args:
+            data(list): sql output for one pdb line
+
+        Returns:
+            str: formatted atom name
+        """
+        name = data[1]
+        lname = len(name)
+        if lname in (1, 4):
+            name = '{:^4}'.format(name)
+        elif lname == 2:
+            # it is a problem when element is not avaiable
+            if not self.no_extra:
+                if name == data[12]:  # name == element
+                    name = '{:<4}'.format(name)
+                else:
+                    name = '{:^4}'.format(name)
+            else:
+                name = '{:^4}'.format(name)
+                warnings.warn(
+                    f'Atom "{name}"" is assumed as one-letter atom name, '
+                    f'which starts at column 14 in a PDB line.')
+        else:
+            if name[0] in '0123456789':
+                name = '{:<4}'.format(name)
+            else:
+                name = '{:>4}'.format(name)
+        return name
+
+    @staticmethod
+    def _format_xyz(i):
+        """Format PDB coordinations x,y or z value.
+
+        Note: PDB has a fixed 8-column space for x,y or z value.
+            Thus the value should be in the range of (-1e7, 1e8).
+
+        Args:
+            i(float): PDB coordinations x, y or z.
+
+        Raises:
+            ValueError: Exceed the range of (-1e7, 1e8)
+
+        Returns:
+            str: formated x, y or z value.
+        """
+
+        if i >= 1e8 - 0.5 or i <= -1e7 + 0.5:
```

Comment:
the second condition is strange no ? 
cause you'll raise a warning for anything lower than 0.5 ? 
I don't quite get the +/- 0.5 actually

Reply:
PDB coordination must be in range of (-1e7, 1e8) due to the fixed number of pdb columns (8 columns). But it can only be in the range of (-1e7 + 0.5, 1e8 - 0.5) to make sure the total number of characters in the coordination value is not larger than 8, otherwise rounding of the value will lead to a need of 9 columns.

Reply:
ok I didn't read it right sorry :)